moveit_simple/src/moveit_simple.cpp

```diff
@@ -219,9 +218,70 @@ bool Robot::getJointSolution(const Eigen::Affine3d &pose, double timeout,
   return getIK(pose, local_seed, joint_point, timeout);
 }
 
+/*------------------------------------------------------------------------------------------------------------------------------*/
+
+/*ISSUE #20 -- getJointSolution() WITH CUSTOM TOOL FRAME FOR SOLVING IK*/
+bool Robot::getJointSolution(const Eigen::Affine3d &pose, const std::string& custom_tool_frame,
+                             double timeout, const std::vector<double> & seed,
+                             std::vector<double> & joint_point) const
+{
+
+  /*Transform Target/Goal Point from custom tool frame to moveit_end_link*/
+  const Eigen::Affine3d custom_frame_goal_pose = 
+                        transformToolToEndLink(pose, custom_tool_frame);
+
+  std::lock_guard<std::recursive_mutex> guard(m_);
+
+  std::vector<double> local_seed = seed;
+  if (seed.empty())
+  {
+    ROS_INFO_STREAM("Empty seed passed to getJointSolution, using current state");
+    local_seed =  getJointState();
+  }
+  return getIK(custom_frame_goal_pose, local_seed, joint_point, timeout);
+}
+
+/* Returns the goal pose transformed to moveit_end_link from Custom tool frame*/
+Eigen::Affine3d Robot::transformToolToEndLink(const Eigen::Affine3d &target_pose,
```

Comment:
If it doesn't lead to too many branches, I always recommend returning from a single point in a function.  This function returns from two different places.  This makes it hard to debug.  Say you wanted to log the return of this method, you would have to track down all the `return` statements and add log messages to each.

Reply:
Ok, yeah that makes sense.. but in that case, would you want me to modify some of the other functions that are designed in a similar way? (Ex.: lookupTrajectoryPoint(...)